src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const getObjectCopy = { ...state };
+  const result = [];
+
+  for (let objectIndex = 0; objectIndex < actions.length; objectIndex++) {
+    const getArraysObject = actions[objectIndex];
+
+    if (actions[objectIndex].type === 'addProperties') {
+      for (const key in getArraysObject.extraData) {
+        getObjectCopy[key] = getArraysObject.extraData[key];
+      }
+      result.push({ ...getObjectCopy });
+    } else if (getArraysObject.type === 'removeProperties') {
+      for (let keyIndex = 0; keyIndex < getArraysObject.keysToRemove.length;
+        keyIndex++) {
+        delete getObjectCopy[getArraysObject.keysToRemove[keyIndex]];
+      }
+      result.push({ ...getObjectCopy });
+    } else if (getArraysObject.type === 'clear') {
+      for (const key in getObjectCopy) {
+        delete getObjectCopy[key];
+      }
```

Comment:
In this case you should use swith statment. Pasha wrote about it in this tread: [here](https://mateacademy.slack.com/archives/C02CSGP1KD1/p1633446273059400)